File: exercism/typescript/armstrong-numbers/armstrong-numbers.ts
export class ArmstrongNumbers {
    public static isArmstrongNumber = (num: number): boolean => {
        const numInString:string = String(num);
        const numberOfDigits:number = numInString.length;
        return num === Array.from(numInString).map(Number).reduce((accumulator, currentValue) => accumulator + currentValue ** numberOfDigits, 0);
    }
}

File: exercism/dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  num _armstrongNumber(int number) {
    return '$number'
        .split("")
        .map(int.parse)
        .map((digit) => pow(digit, '$number'.length))
        .fold(0, (previousValue, value) => previousValue + value);
  }

  bool isArmstrongNumber(int number) => number == _armstrongNumber(number);
}
